Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -497,12 +497,18 @@ private bool SignBlock(List<StakeTx> stakeTxes, Block block, ChainedBlock pindex
             foreach (StakeTx coin in setCoins)
             {
                 this.logger.LogTrace("Trying UTXO from address '{0}', output amount {1}...", coin.Address.Address, coin.TxOut.Value);
-                int maxStakeSearchInterval = 60;
                 bool fKernelFound = false;
+                long minimumAllowedTime = pindexBest.Header.Time + 1;
 
-                for (uint n = 0; n < Math.Min(nSearchInterval, maxStakeSearchInterval) && !fKernelFound; n++)
+                for (uint n = 0; (n < nSearchInterval) && !fKernelFound; n++)
                 {
                     uint txTime = txNew.Time - n;
+
+                    // Once we reach previous block time + 1, we can't go any lower
+                    // because it is required that the block time is greater than the previous block time.
+                    if (txTime < minimumAllowedTime)
```

Comment:
This is a change in functionality, its interesting.
Implications a search will last much longer then 60 iterations.
That means the the current output is probably used for longer periods, while before it would move on to the next output to find a stake after 60 attempts.

Note  this line above (was not taken from stratisx) higher value outputs are selected first.
>setCoins = setCoins.OrderByDescending(o => o.TxOut.Value).ToList();

Reply:
More change in functionality coming in following PR as well.
Yes, previously the backward search was limited to 60 seconds, now I'm allowing unlimited, but this is again limited by the timestamp of the latest block. But according to my current understanding of the consensus rules, there is no 60 seconds limit for backward search on consensus level, so this should be valid.

Yes, I know about the sort, it is minor optimization preventing doing extra work. Very tiny though.

Reply:
ok yeah I think there is no limitation in consensus about backward search just make a mental point this is different in case we come up with issues.

Maybe we can keep a list of logic that is different from X in just in case.

Reply:
I don't really understand here. If not for optimization then for what? The only reason could be optimization for that sort. Yes with sorted list , you have greater chance to pick the right UTXO earlier instead of later. This saves some work, not much though, but still good.

The code will diverge from X very quickly, esp. with my next PR, so I don't know, I think it is important to document all consensus requirements, not really the implementation. This should not be like Bitcoin.